### PR TITLE
Warn on unset exec type

### LINKED ISSUE #162: Warn on missing `#[custom(msg=...)]` or `ExecC` on trait
We want a warning message on interfaces not allowing for custom message specification. The `#[custom(msg=cosmwasm_std::Empty)]` should explicitly shut down this warning.

### PR DESCRIPTION
**Pull request by jawoznia**
close #162

### CODE REVIEW

File: sylvia-derive/src/message.rs [hunk 1]
```diff
@@ -212,6 +212,17 @@ impl<'a> EnumMessage<'a> {
 
         let resp_type = match associated_exec {
             Some(exec) if !custom.has_msg() => exec,
+            None if !custom.has_msg() => {
+                // This will show only on +nightly according to:
+                // https://docs.rs/proc-macro-error/latest/proc_macro_error/macro.emit_warning.html
+                emit_warning!(
+                    source.span(), "The trait is missing the custom message definition, and returning any chain-specific custom messages on its implementation will not be possible.";
+                    note = "Consider adding the `type ExecC;` associated message in the trait definition, and then use it in the execution responses return type: `Result<Response<Self::ExecC>, _>`";
+                    note = "If it is intended and you don't want to allow executing custom messages from this trait, add the `#[sv::custom(cosmwasm_std::Empty)]` attribute on the trait.";
```
Comment (hashedone) [inline sylvia-derive/src/message.rs:221]: Slight typo in second note:

```suggestion
                    note = "If it is intended and you don't want to allow executing custom messages from this trait, add the `#[sv::custom(msg=cosmwasm_std::Empty)]` attribute on the trait.";
```

I know it is what was in task, but it is incorrect.

**Review by hashedone — APPROVED** [review]
Fix failing test (already discused on talk).

Fix warning message.

File: sylvia/examples/generics.rs [hunk 2]
```diff
@@ -9,9 +9,14 @@ where
     Msg: std::fmt::Debug + PartialEq + Clone + schemars::JsonSchema,
 {
     type Error;
+    type ExecC: CustomMsg;
```
Comment (hashedone) [inline sylvia/examples/generics.rs:12]: The idea of this test is that the interface is generic over `Msg` being basically `ExecC`.

Here it should be `#[sv::custom(msg=Msg)]`.

However it doesn't really help with this strange fail.

**Review by hashedone — COMMENTED** [review]
Previosu comment still not fixed, generics comment.

File: sylvia/examples/generics.rs (hunk 2 above)
Comment (jawoznia) [reply sylvia/examples/generics.rs:12]: yes, and as warning would show only on nightly it is for now pointless to merge it.
This PR will be merged once we figure out how to properly approach it

**Comment by jawoznia** [comment]
Closing. We should come back to this issue some time in the future and figure out a way to emit warning (visible on stable) without failing the `tarpaulin`

**Pull request closed without merging**